cns/cnsclient/cnsclient.go

```diff
@@ -198,3 +198,111 @@ func (cnsClient *CNSClient) DeleteHostNCApipaEndpoint(networkContainerID string)
 
 	return nil
 }
+
+// RequestIPAddress calls the requestIPAddress in CNS
+func (cnsClient *CNSClient) RequestIPAddress(orchestratorContext []byte) (*cns.GetNetworkContainerResponse, error) {
+	var (
+		err      error
+		response *cns.GetNetworkContainerResponse
+	)
+
+	defer func() {
+		if err != nil {
+			cnsClient.ReleaseIPAddress(orchestratorContext)
+		}
+	}()
+
+	var body bytes.Buffer
+
+	httpc := &http.Client{}
+	url := cnsClient.connectionURL + cns.RequestIPConfig
+
+	payload := &cns.GetNetworkContainerRequest{
+		OrchestratorContext: orchestratorContext,
+	}
+
+	err = json.NewEncoder(&body).Encode(payload)
+	if err != nil {
+		log.Errorf("encoding json failed with %v", err)
+		return response, err
+	}
+
+	res, err := httpc.Post(url, contentTypeJSON, &body)
```

Comment:
not sure if err is declared here..can we move res declaration in var and remove ":"

Reply:
updated